doc: wave_gen: Add documentation (#5805)
nrfconnect/sdk-nrf: doc/nrf/libraries/others/wave_gen.rst

.. _wave_gen:

Wave generator
##############

.. contents::
   :local:
   :depth: 2

> The wave generator is a simple library that generates values of wave signals.

Reviewer: ```suggestion
The wave generator is a simple library that generates the value of a wave signal.
```

> The wave signal parameters are defined as :c:struct:`wave_gen_param`.

Reviewer: Is it defined as a struct or in a struct? Honestly not sure.

> The :c:func:`wave_gen_generate_value` generates value of the wave signal in predefined time.

Reviewer: ```suggestion
The :c:func:`wave_gen_generate_value` generates the value of the wave signal at a given time.
```
